app/assets/javascripts/views/history_view.js

```diff
@@ -5,41 +5,81 @@ module.exports = Backbone.View.extend({
   initialize: function() {
     _.bindAll(this, 'addActivities', 'connectionError');
 
-    this.currentStory = {};
-
-    this.$el.html(this.template({id: 'history', name: I18n.t('projects.show.history')}));
+    this.$el.html(this.template({
+      id: 'history',
+      className: 'activity_column',
+      name: I18n.t('projects.show.history')
+    }));
+    this.$el.addClass('history_column');
+    this.$column = this.$('#history');
     this.$loadingSpin = $('.loading-spin');
   },
 
-  addActivities: function(activities, a, b) {
-    this.$el.html('');
-    this.trigger('change:currentStory', this.currentStory);
+  events: {
+    'change:currentStory': 'handleStoryChange',
+    'click a.toggle-column': 'toggle'
+  },
+
+  toggle: function() {
+    this.$el.toggle();
+    this.stopListening(this.currentStory, 'sync');
+  },
+
+  handleStoryChange: function(event) {
+    this.setHistoryTitle(event)
+    this.fetchActivities(event)
+    event.stopPropagation();
+  },
 
-    _.each(activities.models, function(item) {
-      this.addActivity(item.attributes);
-    }, this);
+  addActivities: function(activities) {
+    this.$column.html('');
+    activities.models.map((item) => this.addActivity(item.attributes));
 
     this.$loadingSpin.hide();
     this.$el.show();
   },
 
   addActivity: function(activity) {
-    this.$el.append(this.activityTemplate({
+    this.$column.append(this.activityTemplate({
       action: activity.action,
       changes: activity.subject_changes,
       date: activity.date,
-      user: this.options.users.get(activity.user_id).attributes.name
+      user: this.options.users.get(activity.user_id).get('name')
     }));
   },
 
   setStory: function(story) {
-    this.$loadingSpin.show();
-    this.currentStory = story.attributes;
+    if (!this.currentStory || story.get('id') !== this.currentStory.get('id')) {
+      if (this.currentStory) this.stopListening(this.currentStory, 'sync');
+      this.currentStory = story;
+      this.$loadingSpin.show();
+      this.listenTo(this.currentStory, 'sync', this.fetchActivities);
+    }
+    this.$el.trigger('change:currentStory');
+  },
 
-    story.history.fetch({success: this.addActivities, error: this.connectionError});
+  fetchActivities: function() {
+    this.currentStory.history.fetch({
+      success: this.addActivities,
+      error: this.connectionError
+    });
   },
 
   connectionError: function(res) {
     this.$loadingSpin.hide();
+  },
+
+  setHistoryTitle: function() {
+    const $header = this.$el.find('.toggle-title');
+    let title     = this.currentStory.get('title');
+
+    title = this.ellipses(title);
+    $header.text(I18n.t('projects.show.history') + " '" + title + "'");
```

Comment:
You can use template strings here https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/template_strings